Osiris/Hacks/Aimbot.cpp

```diff
@@ -238,3 +250,48 @@ void Aimbot::run(UserCmd* cmd) noexcept
         }
     }
 }
+
+void Aimbot::autoStop(UserCmd *cmd) noexcept
+{
+    if (!localPlayer)
+        return;
+
+    if (localPlayer->moveType() != MoveType::WALK)
+        return;
+
+    Vector velocity = localPlayer->velocity();
+    velocity.z = 0;
+
+    float speed = velocity.length2D();
+
+    if (speed < 1.f) {
+        cmd->forwardmove = 0.f;
+        cmd->sidemove = 0.f;
+        return;
+    }
+
+    float accel = cvars.accelerate->getFloat();
+    float maxSpeed = cvars.maxSpeed->getFloat();
+
+    float playerSurfaceFriction = 1.0f;
+    float maxAccelSpeed = accel * memory->globalVars->intervalPerTick * maxSpeed * playerSurfaceFriction;
+
+    float wishSpeed{};
+
+    if (speed - maxAccelSpeed <= -1.f) wishSpeed = maxAccelSpeed / (speed / (accel * memory->globalVars->intervalPerTick));
+    else wishSpeed = maxAccelSpeed;
+
+    Vector ndir = (velocity * -1.f).toAngle();
+
+    ndir.y = cmd->viewangles.y - ndir.y;
+    ndir = ndir.fromAngle(ndir);
+
+    cmd->forwardmove = ndir.x * wishSpeed;
+    cmd->sidemove = ndir.y * wishSpeed;
+}
+
+void Aimbot::init() noexcept
+{
+    cvars.accelerate = interfaces->cvar->findVar("sv_accelerate");
+    cvars.maxSpeed = interfaces->cvar->findVar("sv_maxspeed");
```

Comment:
pointer to incomplete class type "Cvar" is not allowed

Reply:
#include "../SDK/Cvar.h"